### PR TITLE
lint x %in% NA in equals_na_linter

### LINKED ISSUE #2088: throw equals_na_linter() for x %in% NA
_Originally posted by @AshesITR in https://github.com/r-lib/lintr/pull/2084#discussion_r1296315374_

Originally suggested for inclusion in `scalar_in_linter()`, but agreed it makes more sense to come from `equals_na_linter()`.

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #2088

In writing this I had three follow-up questions

 1. Should we customize the lint message for `%in%`/`==`/`!=` (yes)
 2. Should we lint `x %in% c(NA, y)` to encourage `is.na(x) | x %in% y` (I think no)
 3. Should we lint `NA %in% x` (should be `anyNA(x)`; I am less sure, maybe it could go under `any_is_na_linter()`?)

### CODE REVIEW

File: man/conjunct_test_linter.Rd [hunk 1]
```diff
@@ -44,10 +44,7 @@ lint(
 )
 
 lint(
-  text = "dplyr::filter(
-    mtcars,
-    mpg > 20 & vs == 0
-  )",
+  text = "dplyr::filter(mtcars, mpg > 20 & vs == 0)",
```
Comment (MichaelChirico) [inline man/conjunct_test_linter.Rd:47]: didn't `/document` the previous PR before merging

File: tests/testthat/test-equals_na_linter.R [hunk 2]
```diff
@@ -56,3 +56,7 @@ test_that("equals_na_linter blocks disallowed usages in edge cases", {
   # correct line number for multiline code
   expect_lint("x ==\nNA", list(line_number = 1L, column_number = 1L, ranges = list(c(1L, 4L))), linter)
 })
+
+test_that("x %in% NA is caught", {
+  expect_lint("x %in% NA", "Use is.na for comparisons to NA", equals_na_linter())
```
Comment (AshesITR) [inline tests/testthat/test-equals_na_linter.R:61]: Why not add this to the existing patrick case?

**Review by AshesITR — COMMENTED** [review]
Re 3.: I Agree it should be linted in the `any_is_na_linter()`

**Comment by AshesITR** [comment]
Also should test that `NA %in% x` doesn't lint here.

File: tests/testthat/test-equals_na_linter.R (hunk 2 above)
Comment (MichaelChirico) [reply tests/testthat/test-equals_na_linter.R:61]: Mainly b/c i was too preoccupied thinking about the 3 things mentioned in the description :)

**Comment by MichaelChirico** [comment]
Follow-ups:

#2113
#2114

**Comment by MichaelChirico** [comment]
> Also should test that NA %in% x doesn't lint here.

Good call -- the logic was broken (`==`/`!=` throw lint for NA on LHS or LHS of the operator, not so for `%in%`). Done now

**Review by AshesITR — APPROVED** [review]
